engine/src/main/java/org/pentaho/di/trans/steps/ldifinput/LDIFInput.java

```diff
@@ -0,0 +1,500 @@
+
+    Object[] r = null;
+
+    boolean sendToErrorRow = false;
+    String errorMessage = null;
+
+    try {
+      // Grab one row
+      Object[] outputRowData = getOneRow();
+      if ( outputRowData == null ) {
+        setOutputDone(); // signal end to receiver(s)
+        return false; // end of data or error.
+      }
+
+      putRow( data.outputRowMeta, outputRowData ); // copy row to output rowset(s);
+
+      if ( meta.getRowLimit() > 0 && data.rownr > meta.getRowLimit() ) { // limit has been reached: stop now.
+        setOutputDone();
+        return false;
+      }
+    } catch ( KettleException e ) {
+      if ( getStepMeta().isDoingErrorHandling() ) {
+        sendToErrorRow = true;
+        errorMessage = e.toString();
+      } else {
+        logError( BaseMessages.getString( PKG, "LDIFInput.ErrorInStepRunning", e.getMessage() ) );
+        setErrors( 1 );
+        stopAll();
+        setOutputDone(); // signal end to receiver(s)
+        return false;
+      }
+      if ( sendToErrorRow ) {
+        // Simply add this row to the error row
+        putError( getInputRowMeta(), r, 1, errorMessage, null, "LDIFInput001" );
+      }
+    }
+    return true;
+  }
+
+  private boolean openNextFile() {
+    try {
+      if ( !meta.isFileField() ) {
+        if ( data.filenr >= data.files.nrOfFiles() ) {
+          // finished processing!
+
+          if ( isDetailed() ) {
+            logDetailed( BaseMessages.getString( PKG, "LDIFInput.Log.FinishedProcessing" ) );
+          }
+          return false;
+        }
+
+        // Is this the last file?
+        data.last_file = ( data.filenr == data.files.nrOfFiles() - 1 );
+        data.file = data.files.getFile( data.filenr );
+
+        // Move file pointer ahead!
+        data.filenr++;
+      } else {
+        data.readrow = getRow(); // Get row from input rowset & set row busy!
+        if ( data.readrow == null ) {
+          if ( isDetailed() ) {
+            logDetailed( BaseMessages.getString( PKG, "LDIFInput.Log.FinishedProcessing" ) );
+          }
+          return false;
+        }
+
+        if ( first ) {
+          first = false;
+
+          data.inputRowMeta = getInputRowMeta();
+          data.outputRowMeta = data.inputRowMeta.clone();
+          meta.getFields( data.outputRowMeta, getStepname(), null, null, this, repository, metaStore );
+
+          // Get total previous fields
+          data.totalpreviousfields = data.inputRowMeta.size();
+
+          // Create convert meta-data objects that will contain Date & Number formatters
+          data.convertRowMeta = data.outputRowMeta.cloneToType( ValueMetaInterface.TYPE_STRING );
+
+          // Check is filename field is provided
+          if ( Utils.isEmpty( meta.getDynamicFilenameField() ) ) {
+            logError( BaseMessages.getString( PKG, "LDIFInput.Log.NoField" ) );
+            throw new KettleException( BaseMessages.getString( PKG, "LDIFInput.Log.NoField" ) );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Exception [KettleException] is explicitly thrown on this line but is caught in the outer catch clause with generic exception on line [398].  Consider adding a catch clause of this specific exception in the outer try statement.